src/internal/components/cartesian-chart/__tests__/left-labels.test.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+import LeftLabels from '../../../../../lib/components/internal/components/cartesian-chart/left-labels';
+import {
+  getVisibleTicks,
+  getLabelBBox,
+} from '../../../../../lib/components/internal/components/cartesian-chart/label-utils';
+import { NumericChartScale } from '../../../../../lib/components/internal/components/cartesian-chart/scales';
+
+jest.mock('../../../../../lib/components/internal/components/cartesian-chart/label-utils', () => ({
+  ...jest.requireActual('../../../../../lib/components/internal/components/cartesian-chart/label-utils'),
+  getVisibleTicks: jest.fn().mockReturnValue([]),
+  getLabelBBox: jest.fn().mockReturnValue({ width: 0, height: 0 }),
+}));
+
+jest.mock('../../../../../lib/components/internal/components/responsive-text', () => ({
+  __esModule: true,
+  default: () => <text>...</text>,
+}));
```

Comment:
Is it really necessary to mock everything here?

The more things you mock, the less authentic the tests become

Reply:
True, but the ResponsiveText relies on the API which needs to be mocked as well. By mocking the component I am just making sure the ResponsiveText and not <text> is getting rendered. The implementation is of no significance.